patterns/java/src/refactoring_guru/patterns/decorator/example/decorators/CompressionDecorator.java

```diff
@@ -0,0 +1,64 @@
+package refactoring_guru.patterns.decorator.example.decorators;
+
+import java.io.*;
+import java.util.Scanner;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import java.util.zip.ZipOutputStream;
+
+public class CompressionDecorator extends DataSourceDecorator {
+    FileDataSource fileDataSource;
+
+    public CompressionDecorator(FileDataSource source) {
+        super(source);
+        this.fileDataSource = source;
+    }
+
+    @Override
+    public void writeData(String data) {
+        super.writeData(data);
+        try {
+            File file = new File(fileDataSource.path + fileDataSource.name);
+            FileOutputStream fos = new FileOutputStream(new File(fileDataSource.path + "archive.zip"));
+            ZipOutputStream zos = new ZipOutputStream(fos);
+            zos.setLevel(5);
+            zos.putNextEntry(new ZipEntry(fileDataSource.name));
+            write(new FileInputStream(file), zos);
+            zos.close();
+            file.delete();
+        } catch (FileNotFoundException ex) {
```

Comment:
Can this be combined into one catch block? (http://docs.oracle.com/javase/7/docs/technotes/guides/language/catch-multiple.html)